pkg/prom/instance/instance.go

```diff
@@ -466,6 +465,15 @@ func (i *Instance) Update(c Config) error {
 		return fmt.Errorf("error applying updated configs to scrape manager: %w", err)
 	}
 
+	sdConfigs := map[string]sd_config.ServiceDiscoveryConfig{}
```

Comment:
lgtm :+1: 

sidenote: do we have a test to verify that this order of operation doesn't change in future?

Reply:
we don't, but I think it'd be pretty hard to add an effective test. We never ran into this in testing because applying them in the wrong order will only fail if the service discovery goroutine just happens to send new targets before the goroutine running `Instance.Update` finishes. 

I guess we could mock everything to enforce call order, but I'm not sure that's worth the extra complexity over just having the comment warning developers about what order things should be applied in.

Reply:
since this is something only agent developers interact with, we should be fine with a comment. 

if we have development docs, maybe we should add a note in relevant development docs

Reply:
:shipit: